content/learn/book/migration-guides/0.6-0.7/_index.md

```diff
@@ -0,0 +1,211 @@
+
+https://github.com/bevyengine/bevy/pull/3812
+
+Instead of using a boolean flag for vsync we switched to using a [`PresentMode`] enum with multiple variants.
+
+```rs
+// 0.6
+App::new()
+    .insert_resource(WindowDescriptor {
+        vsync: false,
+        ..Default::default()
+    })
+
+// 0.7
+App::new()
+    .insert_resource(WindowDescriptor {
+        present_mode: PresentMode::Immediate,
+        ..Default::default()
+    })
+```
+
+<!-- TODO update to 0.7 link -->
+[`PresentMode`]: http://dev-docs.bevyengine.org/bevy/window/enum.PresentMode.html
+
+### Fix mul_vec3 tranformation order
+
+<https://github.com/bevyengine/bevy/pull/3811>
+
+Transforms are now consistently applied in the standard scale -> rotate -> translate. This doesn't require any code changes, but it means SpriteBundle will behave as expected when rotating.
+
+### Use marker components for cameras instead of name strings
+
+<https://github.com/bevyengine/bevy/pull/3635>
+
+TODO
+
+### Remove the config api
+
+<https://github.com/bevyengine/bevy/pull/3633>
+
+TODO
+
+### Cameras now point at RenderTarget rather than Window
+
+<https://github.com/bevyengine/bevy/pull/3412>
+
+This change was made to support rendering to textures. Users working with multiple windows may be affected.
+
+```rs
+// 0.6
+commands.spawn_bundle(PerspectiveCameraBundle {
+    camera: Camera {
+        window: window_id,
+        ..Default::default()
+    },
+    ..Default::default()
+});
+
+// 0.7
+commands.spawn_bundle(PerspectiveCameraBundle {
+    camera: Camera {
+        target: RenderTarget::Window(window_id),
+        ..Default::default()
+    },
+    ..Default::default()
+});
+```
+
+### Implement init_resource for Commands and World
+
+<https://github.com/bevyengine/bevy/pull/3079>
+
+```rs
+#[derive(Default)]
+struct Scoreboard {
+    current_score: u32,
+    high_score: u32,
+}
+
+// 0.6
+commands.insert_resource(Scoreboard::Default());
+
+// 0.7
+commands.init_resource::<Scoreboard>();
+```
+
+### ParamSet for conflicting SystemParam
+
+<https://github.com/bevyengine/bevy/pull/2765>
+
+```rs
+// 0.6
+fn system(
+    mut transforms: QuerySet<(
+        QueryState<&mut Transform, With<Marker>>,
+        QueryState<&Transform>,
+    )>,
+) {
+    for transform in transforms.q1().iter() {
+        // ...
+    }
+}
+// 0.7
+fn system(
+    mut transforms: ParamSet<(
+        Query<&mut Transform, With<Marker>>,
+        Query<&Transform>
+    )>,
+) {
+    for transform in transforms.p1().iter() {
+        // ...
+    }
+}
+```
+
+### Infallabile resource getters
+
+<https://github.com/bevyengine/bevy/pull/4047>
+
+```rs
+// 0.6
+let score = world.get_resource::<Score>().unwrap();
+
+// 0.7
+let score = world.resource::<Score>();
+```
+
+### Events have been moved form bevy_app to bevy_ecs
+
+<https://github.com/bevyengine/bevy/pull/4066>
+
+If you are using the prelude it won't change anything.
+
+```rs
+// 0.6
+use bevy::app::Events;
+
+// 0.7
+use bevy::ecs::event::Events;
+```
```

Comment:
Noticed a typo in "from" but went digging on this issue and saw that this affects more than just the `Events` type.

~~~suggestion
### Event handling types are no longer re-exported from bevy_app

<https://github.com/bevyengine/bevy/pull/4066>

This only affects users who were importing these types directly from `bevy_app` and not through bevy's prelude.

```rs
// 0.6
use bevy::app::{EventId, EventReader, EventWriter, Events, ManualEventReader};

// 0.7
use bevy::ecs::event::{EventId, EventReader, EventWriter, Events, ManualEventReader};
```
~~~